### PR TITLE
warn about master pushes

### LINKED ISSUE #146: Alert if push to master?
We would like to have alert in mattermost channel if someone pushes to master (instead of using PRs and normal PR process). I think this cannot be done with current plugin?

### PR DESCRIPTION
**Pull request by zetaab**
#### Summary

This PR notifies channel only if non-whitelisted user will push to master. Whitelist is needed in order to use normal PR processes.

#### Ticket Link

Fixes #146

### CODE REVIEW

**Comment by zetaab** [comment]
looks like:

Screenshot 2020-01-08 at 11 29 40

**Comment by hanzei** [comment]
Hi @zetaab,

Thanks for the PR. I'm wondering what exactly the use case of this feature is. GitHub already as a setting to disallow pushed to e.g. `master`
Screenshot from 2020-01-08 11-59-52

**Comment by zetaab** [comment]
@hanzei we are using github enterprise and at least for us these rules does not work like should. Also these rules are somehow cleaned sometimes (in updates?). I configured these rules for one repository, but still I can push to master. So I would say that feature is useless.

**Comment by hanzei** [comment]
@zetaab Is it possible that you didn't enable the rules for admins?
Screenshot from 2020-01-08 13-35-36

File: server/utils.go [hunk 1]
```diff
@@ -310,3 +310,21 @@ func getCodeMarkdown(user, repo, repoPath, word, lines string, isTruncated bool)
 	final += "```\n"
 	return final
 }
+
+// stringValue returns string value from pointervalue
```
Comment (jwilander) [inline server/utils.go]:
```suggestion
// stringValue returns the string value from a string pointer
```

File: server/utils.go [hunk 2]
```diff
@@ -310,3 +310,21 @@ func getCodeMarkdown(user, repo, repoPath, word, lines string, isTruncated bool)
...
+func stringValue(s *string) string {
+	if s == nil {
+		return ""
+	}
+	return *s
+}
+
+// contains is checking does array contain single word
```
Comment (jwilander) [inline server/utils.go]:
```suggestion
// contains checks if a string array contains a word
```

**Comment by hanzei** [comment]
Hey @zetaab,

What do you thin about my feedback at https://github.com/mattermost/mattermost-plugin-github/pull/173#issuecomment-572029386? Does this setting help with your use case?

**Comment by hanzei** [comment]
@zetaab given that this PR has been stale for some time and the use case is questionable, I'm going to close this PR. Still, thanks for your PR

**Pull request closed without merging**